src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,189 @@
+    }
+
+    @Test
+    void register_emptyLogin_throwsException() {
+        User user = new User();
+        user.setLogin("");
+        user.setPassword("qwerty1234");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_throwsException() {
+        User user = new User();
+        user.setLogin("Bob1256");
+        user.setPassword(null);
+        user.setAge(25);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_emptyPassword_throwsException() {
+        User user = new User();
+        user.setLogin("Bob1256");
+        user.setPassword("");
+        user.setAge(25);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_loginLessThanMinLength_throwsException() {
+        User user = new User();
+        user.setLogin("Bob");
+        user.setPassword("qwerty1234");
+        user.setAge(25);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordLessThanMinLength_throwsException() {
+        User user = new User();
+        user.setLogin("Bob1256");
+        user.setPassword("qwert");
+        user.setAge(25);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_underageUser_throwsException() {
+        User user = new User();
+        user.setLogin("Alice");
+        user.setPassword("qwerty1234");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_validDetails_ok() {
+        User user = new User();
+        user.setPassword("qwerty123");
+        user.setLogin("Alex1976");
+        user.setAge(27);
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser);
+        User storedUser = storageDao.get(user.getLogin());
+        assertEquals(user, storedUser);
+    }
+
+    @Test
+    void register_invalidAge_throwsException() {
+        User user = new User();
+        user.setLogin("Bob1256");
+        user.setPassword("password");
+        user.setAge(-1);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_exactMinAge_ok() {
+        User user = new User();
+        user.setLogin("Alice123");
+        user.setPassword("qwerty1234");
+        user.setAge(MIN_AGE);
+        User registeredUser = registrationService.register(user);
+        assertEquals(user, registeredUser);
+        User storedUser = storageDao.get(user.getLogin());
+        assertEquals(user, storedUser);
+    }
+
+    @Test
+    void testGetUserByLogin_success() {
```

Comment:
You actually test method register, not storageDao.get(), so this test case is redundant here